## <center>Actividad 3 - Análisis exploratorio de datos + Machine Learning</center>

Estudiantes: Rubén Martínez Gormaz | Samuel Fernando Méndez | Ruoyan Zhang | Lian Salmerón López

## Grupo 6

En esta actividad se pide realizar un análisis profundo sobre un conjunto de datos de un tema de negocio, para llevar a cabo un procesamiento de la información y entrenar un modelo de aprendizaje automático, que consiga predecir los siguientes valores del conjunto de datos.

Para esta activiad se ha escogido el tema del precio de los combustibles en España durante el 2022.

## <center><u>Instalación de librerías</u></center>

Las librerías utilizadas para esta investigación son las siguientes, si no se tiene alguna de ellas, debe ejecutar la siguiente línea.

In [ ]:
## Dejar para librerias nuevas que utilicemos

Una vez instaladas, se importan al proyecto.

In [277]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import os

- La librería <b>pandas</b> se especializa en la manipulación y análisis de datos. Ofrece estructuras de datos y operaciones para manipular tablas numéricas y series temporales. <br>
- La librería <b>numpy</b> ofrece la posibilidad de crear vectores y matrices grandes multidimensionales, junto con una gran colección de funciones matemáticas.<br>
- La libería <b>os</b>  provee una manera versátil de usar funcionalidades dependientes del sistema operativo.

## <center><u>Desarrollo</u></center>

Primero se deben cargar los ficheros descargados, sin embargo éstos se encuentran en un formato diferente (<i>.parquet</i>), para trabajar con ellos de una manera más sencilla, se transforman a documentos <i>.csv</i>.

In [ ]:
# Se realiza en un rango de 0 a 17, ya que existen 17 documentos de datos.
for fichero_parquet in range(0, 17):
    # Si el fichero no supera el noveno documento
    if fichero_parquet > 9:
        parquet = pd.read_parquet('spanish_gas_prices_00000000000'+str(fichero_parquet)+'.parquet')
        parquet.to_csv('spanish_gas_prices_00000000000'+str(fichero_parquet)+'.csv')
        print("hola" + fichero_parquet)
    # Si el fichero supera el noveno documento
    else:
        parquet = pd.read_parquet('spanish_gas_prices_0000000000'+str(fichero_parquet)+'.parquet')
        parquet.to_csv('spanish_gas_prices_0000000000'+str(fichero_parquet)+'.csv')
        print("adios" + fichero_parquet)

Una vez transformados todos los documentos al formato que se desea, se empieza a cargar todos los ficheros.

### Carga de datos

Para cargar los datos que se encuentran en los csv, se cargarán de uno a uno los 17 documentos, para después unirlos en una misma tabla y trabajar con ellos. Por lo tanto se realizarán los filtros necesarios (en este caso obtener el precio de la gasolina para madrid durante el 2022). <br>
Todos los archivos mantienen una misma estructura por lo que se explicarán las características con la ayuda del priemr .csv

## CSV 00

In [78]:
# Se carga el primer csv
df_gasoline_00 = pd.read_csv('D:\IA\spanish_gas_prices_000000000000.csv')

In [79]:
df_gasoline_00.shape

(3043717, 32)

Todos los documentos tendrán 32 columnas, con un número de filas que puede variar.

In [80]:
df_gasoline_00.head()

station_id                              name  \
0        4368  ESTACIÓN DE SERVICIO LOS VIÑEDOS   
1        3468                             CEPSA   
2        7229                            BP OIL   
3        7421                            REPSOL   
4         726                            REPSOL   

                               address                       town  zip_code  \
0       CARRETERA DE LAS MESAS KM. 005  VILLARROBLEDO                  2600   
1  AVENIDA DOCTOR GREGORIO MARAÑÓN, 14  TORREVIEJA                     3180   
2             CARRETERA N-IV KM. 536,7  SEVILLA                       41020   
3                   AU AP-7  224,00  I  TORREMOLINOS                  29620   
4    CARRETERA CAMBADOS-NOGUEIRA KM. 8  VILANOVA DE AROUSA            36620   

   longitude   latitude road_side restriction sender  ... gasoline_98E10  \
0  -2.617083  39.272250         N           P     dm  ...            NaN   
1  -0.687444  37.977278         I           P     dm  ...            NaN   
2  -5.972472  37.337278         N           P     dm  ...            NaN   
3  -4.527250  36.613611         I           P     dm  ...            NaN   
4  -8.756111  42.536417         D           P     dm  ...            NaN   

   diesel_A  diesel_B  diesel_premium bioetanol biodiesel lpg cng  lng  \
0     1.879     1.459           1.969       NaN       NaN NaN NaN  NaN   
1     1.899       NaN           1.965       NaN       NaN NaN NaN  NaN   
2     1.859       NaN           1.939       NaN       NaN NaN NaN  NaN   
3     1.929       NaN           2.019       NaN       NaN NaN NaN  NaN   
4     1.979     1.619           2.019       NaN       NaN NaN NaN  NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN  

[5 rows x 32 columns]

La tabla resultante resulta ser demasiado grande para visualizar, por lo tanto se explicarán todas las columnas que aparecen en el documento.

In [81]:
df_gasoline_00.columns

Index(['station_id', 'name', 'address', 'town', 'zip_code', 'longitude',
       'latitude', 'road_side', 'restriction', 'sender', 'schedule',
       'region_id', 'province_id', 'municipality_id', 'region_name',
       'province_name', 'municipality_name', 'date', 'gasoline_95E5',
       'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5',
       'gasoline_98E10', 'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol',
       'biodiesel', 'lpg', 'cng', 'lng', 'hydrogen'],
      dtype='object')

### Datos

A continunación se explicarán todos los datos del csv
<ul>
    <li>station_id - El identificador de las estaciones de servicio.</li>
    <li>name - Nombre de la estación de servicio.</li>
    <li>address - Dirección de dónde se encuentra la estación de servicio.</li>
    <li>town - Ciudad en la que se encuentra la estación de servicio.</li>
    <li>zip_code - Código postal de la región en la que se halla la la estación de servicio.</li>
    <li>longitude - Coordenada geográfica que equivale al eje X para indicar la posición de la estación de servicio.</li>
    <li>latitude - Coordenada geográfica que equivale al eje Y para indicar la posición de la estación de servicio.</li>
    <li>road_side - El lado de la carretera al que se encuentra la estación de servicio (N) Ninguna, (D) Derecha, (I) Izquierda.</li>
    <li>restriction - Indica si la estación es de acceso público (P) o se encuentra restringido (R).</li>
    <li>sender - El distribuidor, pudiendo ser un operador mayorista (OM) o un distribuidor menor (dm).</li>
    <li>schedule - Programa de horarios en los que se encuentra disponible la estación de servicio.</li>
    <li>region_id - Identificador de la región a la que pertenece la estación.</li>
    <li>province_id - Identificador de la provincia a la que pertenece la estación.</li>
    <li>municipality_id - Identificador del municipio al que pertenece la estación.</li>
    <li>region_name - Nombre de la región a la que pertenece la estación.</li>
    <li>province_name - Nombre de la provincia a la que pertence la estación.</li>
    <li>municipality_name - Nombre del municipio al que pertence la estación.</li>
    <li>date - Fecha en la que se tomaron los datos de los diferentes carburantes.</li>
    <li>gasoline_95E5 - Precio de la gasolina 95E5 para una cierta fecha.</li>
    <li>gasoline_95E5_premium - Precio de la gasolina 95E5 premium para una cierta fecha.</li>
    <li>gasoline_95E10 - Precio de la gasolina 95E10 para una cierta fecha.</li>
    <li>gasoline_98E5 - Precio de la gasolina 98E5 para una cierta fecha.</li>
    <li>gasoline_98E10 - Precio de la gasolina 98E10 para una cierta fecha.</li>
    <li>diesel_A - Precio del diesel A para una cierta fecha.</li>
    <li>diesel_B - Precio del diesel B para una cierta fecha.</li>
    <li>diesel_premium - Precio del diesel premium para una cierta fecha.</li>
    <li>bioetanol - Precio del bioetanol para una cierta fecha.</li>
    <li>biodiesel - Precio del biodiesel para una cierta fecha.</li>
    <li>lpg - Precio para el lpg (Gas licuado del petróleo) para una cierta fecha.</li>
    <li>cng - Precio para el cng (Gas natural comprimido) para una cierta fecha.</li>
    <li>lng - Precio para el lng (Gas natural licuado) para una cierta fecha.</li>
    <li>hydrogen - Precio del hidrógeno para una cierta fecha.</li>
</ul>

Una vez explicada los diferentes datos que se pueden encontrar en cada csv, se va a crear una tabla con solamente las variables importantes para esta investigación.

In [82]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_00_v1 = df_gasoline_00[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5',
       'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5',
       'gasoline_98E10', 'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol',
       'biodiesel', 'hydrogen']].copy()

In [83]:
df_gasoline_00_v1.head()

station_id           region_name   municipality_name        date  \
0        4368    Castilla la Mancha       Villarrobledo  2022-05-31   
1        3468  Comunidad Valenciana          Torrevieja  2022-05-31   
2        7229             Andalucia             Sevilla  2022-05-31   
3        7421             Andalucia        Torremolinos  2022-05-31   
4         726               Galicia  Vilanova de Arousa  2022-06-03   

   gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0          1.959                    NaN             NaN          2.079   
1          2.044                  2.104             NaN          2.176   
2          1.999                    NaN             NaN          2.149   
3          2.089                    NaN             NaN          2.239   
4          2.159                    NaN             NaN          2.259   

   gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
0             NaN     1.879     1.459           1.969        NaN        NaN   
1             NaN     1.899       NaN           1.965        NaN        NaN   
2             NaN     1.859       NaN           1.939        NaN        NaN   
3             NaN     1.929       NaN           2.019        NaN        NaN   
4             NaN     1.979     1.619           2.019        NaN        NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

Una vez creada, se van a aplicar los filtros explicados con anterioridad, buscar datos en Madrid durante el año 2022.

In [84]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_00 = df_gasoline_00_v1[(df_gasoline_00_v1["date"].str.contains("2022"))]

In [85]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_00 = df_gasoline2022_00[(df_gasoline2022_00["region_name"]=="Madrid")]

In [86]:
df_madrid2022_00.head()

station_id region_name       municipality_name        date  \
13         4312      Madrid                  Madrid  2022-06-03   
15         3201      Madrid                  Madrid  2022-06-01   
77         3222      Madrid  Torrejón de la Calzada  2022-03-04   
599        4600      Madrid                  Madrid  2022-03-25   
702        4364      Madrid             Navacerrada  2022-02-25   

     gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
13           2.169                  2.229             NaN            NaN   
15           2.099                    NaN             NaN          2.269   
77           1.739                  1.799             NaN            NaN   
599          1.889                    NaN             NaN          2.029   
702          1.664                  1.744             NaN          1.796   

     gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
13              NaN     2.005       NaN           2.085        NaN        NaN   
15              NaN     1.959       NaN           2.049        NaN        NaN   
77              NaN     1.655       NaN           1.735        NaN        NaN   
599             NaN     1.919       NaN           1.999        NaN        NaN   
702             NaN     1.549       NaN           1.629        NaN        NaN   

     hydrogen  
13        NaN  
15        NaN  
77        NaN  
599       NaN  
702       NaN

La tabla resultante resulta ser mucho más pequeña comparada con la gran cantidad de datos hallados en el documento original. <br>
Todas las tablas que resulten de la carga y filtrado de los ficheros csv originales se verán reducidos en gran número, que luego se unirán bajo una misma tabla para su posterior análisis.

Una vez explicada la organización seguida para la carga de los diferentes ficheros, las siguientes se realizarán de la misma manera, por lo que no se volverá a explicar lo detallado con anterioridad.

## CSV 01

In [72]:
df_gasoline_01 = pd.read_csv('D:\IA\spanish_gas_prices_000000000001.csv')

In [73]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_01_v1 = df_gasoline_01[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5',
       'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5',
       'gasoline_98E10', 'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol',
       'biodiesel', 'hydrogen']].copy()

In [74]:
df_gasoline_01_v1.head()

station_id         region_name  municipality_name        date  \
0         723             Galicia         Pontevedra  2022-05-31   
1        2059            Cataluña           Badalona  2022-06-01   
2        2573            Baleares  Palma de Mallorca  2022-05-31   
3        1356              Aragón          Calatayud  2022-05-31   
4        4859  Castilla la Mancha          Mirabueno  2022-06-02   

   gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0          2.069                    NaN             NaN          2.199   
1          1.878                    NaN             NaN          1.995   
2          2.099                    NaN             NaN          2.249   
3          1.899                    NaN             NaN            NaN   
4          2.099                    NaN             NaN            NaN   

   gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
0             NaN     1.939       NaN           1.989        NaN        NaN   
1             NaN     1.837       NaN           1.867        NaN        NaN   
2             NaN     1.955       NaN           2.025        NaN        NaN   
3             NaN     1.799       NaN           1.869        NaN        NaN   
4             NaN     1.959       NaN           2.009        NaN        NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [75]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_01 = df_gasoline_01_v1[df_gasoline_01_v1["date"].str.contains("2022")]

In [76]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_01 = df_gasoline2022_01[(df_gasoline2022_01["region_name"]=="Madrid")]

In [77]:
df_madrid2022_01.head()

station_id region_name municipality_name        date  gasoline_95E5  \
12          4563      Madrid            Madrid  2022-06-02          2.139   
314         9441      Madrid            Madrid  2022-02-28          1.679   
930         4711      Madrid            Madrid  2022-01-06          1.579   
1194        3091      Madrid     Navas del Rey  2022-03-05          1.724   
1401        4586      Madrid            Madrid  2022-04-24          1.869   

      gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  gasoline_98E10  \
12                      NaN             NaN            NaN             NaN   
314                     NaN             NaN          1.829             NaN   
930                     NaN             NaN          1.704             NaN   
1194                    NaN             NaN          1.844             NaN   
1401                    NaN             NaN          2.019             NaN   

      diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  hydrogen  
12       1.979       NaN           2.069        NaN        NaN       NaN  
314      1.559       NaN           1.589        NaN        NaN       NaN  
930      1.429       NaN           1.497        NaN        NaN       NaN  
1194     1.649       NaN           1.719        NaN        NaN       NaN  
1401     1.919       NaN           2.009        NaN        NaN       NaN

## CSV 02

In [87]:
df_gasoline_02 = pd.read_csv('D:\IA\spanish_gas_prices_000000000002.csv')

In [88]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_02_v1 = df_gasoline_02[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5',
       'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5',
       'gasoline_98E10', 'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol',
       'biodiesel', 'hydrogen']].copy()

In [89]:
df_gasoline_02_v1.head()

station_id           region_name municipality_name        date  \
0        2994              Cataluña            Mataró  2022-06-03   
1        2560              Cataluña             Olost  2022-06-03   
2       12323  Comunidad Valenciana      Alcúdia (l')  2022-06-03   
3        6839             Andalucia          Tabernas  2022-06-03   
4        4673                Madrid  Sevilla la Nueva  2022-06-03   

   gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0          2.159                    NaN             NaN          2.319   
1          2.119                    NaN             NaN            NaN   
2          1.889                    NaN             NaN            NaN   
3          2.089                    NaN             NaN          2.239   
4          2.104                    NaN             NaN          2.224   

   gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
0             NaN     1.999       NaN           2.069        NaN        NaN   
1             NaN     1.969       NaN           2.029        NaN        NaN   
2             NaN     1.849       NaN           1.909        NaN        NaN   
3             NaN     1.949       NaN           2.039        NaN        NaN   
4             NaN     1.984       NaN           2.054        NaN        NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [90]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_02 = df_gasoline_02_v1[df_gasoline_02_v1["date"].str.contains("2022")]

In [91]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_02 = df_gasoline2022_02[(df_gasoline2022_02["region_name"]=="Madrid")]

In [92]:
df_madrid2022_02.head()

station_id region_name           municipality_name        date  \
4          4673      Madrid            Sevilla la Nueva  2022-06-03   
24         4358      Madrid  San Sebastián de los Reyes  2022-06-01   
242        2928      Madrid                  Villamanta  2022-03-16   
255        4531      Madrid                      Madrid  2022-01-25   
418        4578      Madrid         Buitrago del Lozoya  2022-04-18   

     gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
4            2.104                    NaN             NaN          2.224   
24           2.099                    NaN             NaN          2.249   
242          1.889                    NaN             NaN          2.031   
255          1.589                    NaN             NaN          1.729   
418          1.839                    NaN             NaN          1.969   

     gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
4               NaN     1.984       NaN           2.054        NaN        NaN   
24              NaN     1.949       NaN           2.039        NaN        NaN   
242             NaN     1.879       NaN           1.969        NaN        NaN   
255             NaN     1.479       NaN           1.569        NaN        NaN   
418             NaN     1.869       NaN           1.951        NaN        NaN   

     hydrogen  
4         NaN  
24        NaN  
242       NaN  
255       NaN  
418       NaN

## CSV 03

In [93]:
df_gasoline_03 = pd.read_csv('D:\IA\spanish_gas_prices_000000000003.csv')

In [94]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_03_v1 = df_gasoline_03[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5',
       'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5',
       'gasoline_98E10', 'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol',
       'biodiesel', 'hydrogen']].copy()

In [95]:
df_gasoline_03_v1.head()

station_id           region_name  \
0       13063  Comunidad Valenciana   
1        7413             Andalucia   
2        9798            Rioja (La)   
3        2697              Baleares   
4        4093  Comunidad Valenciana   

                            municipality_name        date  gasoline_95E5  \
0  Castellón de la Plana/Castelló de la Plana  2022-06-01          2.079   
1                                     Sevilla  2022-06-03          2.129   
2                                     Logroño  2022-06-03          1.995   
3                       Santa Eulalia del Río  2022-05-31          2.139   
4                                    Valencia  2022-06-01          2.119   

   gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  gasoline_98E10  \
0                    NaN             NaN          2.221             NaN   
1                    NaN             NaN          2.249             NaN   
2                    NaN             NaN          2.091             NaN   
3                    NaN             NaN            NaN             NaN   
4                    NaN             NaN          2.229             NaN   

   diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  hydrogen  
0     1.899       NaN           1.989        NaN        NaN       NaN  
1     1.969       NaN           2.069        NaN        NaN       NaN  
2     1.824       NaN           1.851        NaN        NaN       NaN  
3     1.985       NaN           2.085        NaN        NaN       NaN  
4     1.935       NaN           2.025        NaN        NaN       NaN

In [96]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_03 = df_gasoline_03_v1[df_gasoline_03_v1["date"].str.contains("2022")]

In [97]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_03 = df_gasoline2022_03[(df_gasoline2022_03["region_name"]=="Madrid")]

In [98]:
df_madrid2022_03.head()

station_id region_name           municipality_name        date  \
21         4358      Madrid  San Sebastián de los Reyes  2022-06-02   
50        11722      Madrid                      Madrid  2022-01-23   
182        2916      Madrid                      Madrid  2022-02-16   
219        3208      Madrid       Pezuela de las Torres  2022-04-10   
619        4609      Madrid                    Móstoles  2022-03-25   

     gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
21           2.119                    NaN             NaN          2.269   
50           1.629                    NaN             NaN          1.779   
182          1.659                    NaN             NaN          1.769   
219          1.769                    NaN             NaN            NaN   
619          1.859                    NaN             NaN            NaN   

     gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
21              NaN     1.969       NaN           2.059        NaN        NaN   
50              NaN     1.519       NaN           1.569        NaN        NaN   
182             NaN     1.549       NaN           1.599        NaN        NaN   
219             NaN     1.809       NaN           1.859        NaN        NaN   
619             NaN     1.929       NaN           1.999        NaN        NaN   

     hydrogen  
21        NaN  
50        NaN  
182       NaN  
219       NaN  
619       NaN

## CSV 04

In [99]:
df_gasoline_04 = pd.read_csv('D:\IA\spanish_gas_prices_000000000004.csv')

In [100]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_04_v1 = df_gasoline_04[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5',
       'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5',
       'gasoline_98E10', 'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol',
       'biodiesel', 'hydrogen']].copy()

In [101]:
df_gasoline_04_v1.head()

station_id         region_name  municipality_name        date  \
0        8500             Galicia               Vigo  2022-05-31   
1         529             Galicia            Carnota  2022-06-03   
2        5181  Castilla la Mancha  Valverde de Júcar  2022-06-01   
3        1502              Aragón          Calatorao  2022-05-31   
4        6648           Andalucia      Nueva Carteya  2022-06-02   

   gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0          2.099                    NaN             NaN          2.209   
1          2.169                  2.199             NaN            NaN   
2          2.070                    NaN             NaN            NaN   
3          2.059                    NaN             NaN            NaN   
4          1.949                    NaN             NaN            NaN   

   gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
0             NaN     1.939       NaN           1.999        NaN        NaN   
1             NaN     1.969       NaN           1.979        NaN        NaN   
2             NaN     2.000     1.600           2.060        NaN        NaN   
3             NaN     1.899       NaN           1.989        NaN        NaN   
4             NaN     1.859     1.459           1.889        NaN        NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [102]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_04 = df_gasoline_04_v1[df_gasoline_04_v1["date"].str.contains("2022")]

In [103]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_04 = df_gasoline2022_04[(df_gasoline2022_04["region_name"]=="Madrid")]

In [104]:
df_madrid2022_04.head()

station_id region_name        municipality_name        date  \
10         8464      Madrid  Villanueva de la Cañada  2022-06-02   
40        11025      Madrid                   Madrid  2022-06-01   
63        11344      Madrid                   Madrid  2022-04-03   
77         2928      Madrid               Villamanta  2022-05-02   
310        8058      Madrid   Velilla de San Antonio  2022-05-06   

     gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
10           2.079                    NaN             NaN          2.219   
40           2.124                  2.184             NaN            NaN   
63           1.829                    NaN             NaN          1.969   
77           1.909                    NaN             NaN          2.051   
310          1.919                    NaN             NaN          2.051   

     gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
10              NaN     1.939       NaN           2.019        NaN        NaN   
40              NaN     1.959       NaN           2.039        NaN        NaN   
63              NaN     1.869       NaN           1.949        NaN        NaN   
77              NaN     1.989       NaN           2.079        NaN        NaN   
310             NaN     1.985       NaN           2.065        NaN        NaN   

     hydrogen  
10        NaN  
40        NaN  
63        NaN  
77        NaN  
310       NaN

## CSV 05

In [105]:
df_gasoline_05 = pd.read_csv('D:\IA\spanish_gas_prices_000000000005.csv')

In [106]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_05_v1 = df_gasoline_05[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5',
       'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5',
       'gasoline_98E10', 'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol',
       'biodiesel', 'hydrogen']].copy()

In [107]:
df_gasoline_05_v1.head()

station_id           region_name  municipality_name        date  \
0         418             Cantabria  Campoo de Enmedio  2022-06-01   
1        7266             Andalucia              Écija  2022-06-03   
2        5147    Castilla la Mancha             Cuenca  2022-06-01   
3        3269              Cataluña            Tortosa  2022-06-02   
4        4030  Comunidad Valenciana             Cheste  2022-06-03   

   gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0          2.019                    NaN             NaN          2.189   
1          2.119                    NaN             NaN            NaN   
2          2.089                    NaN             NaN          2.239   
3          2.129                    NaN             NaN          2.279   
4          1.840                    NaN             NaN          1.900   

   gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
0             NaN     1.899     1.417           1.969        NaN        NaN   
1             NaN     1.929     1.609           1.999        NaN        NaN   
2             NaN     1.949       NaN           2.039        NaN        NaN   
3             NaN     1.959       NaN           2.009        NaN        NaN   
4             NaN     1.790     1.390           1.830        NaN        NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [108]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_05 = df_gasoline_05_v1[df_gasoline_05_v1["date"].str.contains("2022")]

In [109]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_05 = df_gasoline2022_05[(df_gasoline2022_05["region_name"]=="Madrid")]

In [110]:
df_madrid2022_05.head()

station_id region_name      municipality_name        date  gasoline_95E5  \
13         4661      Madrid  Rozas de Madrid (Las)  2022-06-02          2.109   
28         3134      Madrid                 Getafe  2022-06-02          2.099   
352       11746      Madrid                 Madrid  2022-04-14          1.819   
451        4568      Madrid                Loeches  2022-04-08          1.799   
604       12333      Madrid                  Pinto  2022-02-10          1.639   

     gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  gasoline_98E10  \
13                     NaN             NaN          2.279             NaN   
28                     NaN             NaN          2.241             NaN   
352                    NaN             NaN          1.959             NaN   
451                    NaN             NaN          1.949             NaN   
604                    NaN             NaN          1.779             NaN   

     diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  hydrogen  
13      1.979       NaN           2.019        NaN        NaN       NaN  
28      1.979       NaN           2.069        NaN        NaN       NaN  
352     1.849     1.522           1.899        NaN        NaN       NaN  
451     1.859       NaN           1.909        NaN        NaN       NaN  
604     1.539       NaN           1.629        NaN        NaN       NaN

## CSV 06

In [3]:
df_gasoline_06 = pd.read_csv('D:\IA\spanish_gas_prices_000000000006.csv')

In [4]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_06_v1 = df_gasoline_06[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5', 'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5', 'gasoline_98E10', 
                                    'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol', 
                                    'biodiesel', 'hydrogen']].copy()

In [5]:
df_gasoline_06_v1.head()

station_id region_name        municipality_name        date  gasoline_95E5  \
0         947    Asturias                   Valdés  2022-06-02          2.149   
1        1801    Cataluña                     Gavà  2022-06-01          2.159   
2         649     Galicia   Santiago de Compostela  2022-06-01          2.249   
3        4685      Madrid  San Fernando de Henares  2022-06-03          2.103   
4       13694    Canarias      San Miguel de Abona  2022-05-31          1.489   

   gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  gasoline_98E10  \
0                    NaN             NaN          2.259             NaN   
1                    NaN             NaN          2.309             NaN   
2                    NaN             NaN          2.305             NaN   
3                    NaN             NaN          2.223             NaN   
4                    NaN             NaN          1.604             NaN   

   diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  hydrogen  
0     1.919       NaN           1.959        NaN        NaN       NaN  
1     1.979       NaN           2.049        NaN        NaN       NaN  
2     1.979       NaN           1.999        NaN        NaN       NaN  
3     1.983       NaN           2.043        NaN        NaN       NaN  
4     1.484       NaN           1.619        NaN        NaN       NaN

In [6]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_06 = df_gasoline_06_v1[df_gasoline_06_v1["date"].str.contains("2022")]

In [7]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_06 = df_gasoline2022_06[(df_gasoline2022_06["region_name"]=="Madrid")]

In [8]:
df_madrid2022_06.head()

station_id region_name        municipality_name        date  \
3          4685      Madrid  San Fernando de Henares  2022-06-03   
219       12119      Madrid                    Pinto  2022-04-13   
254        3159      Madrid                   Madrid  2022-01-23   
562        3178      Madrid                   Algete  2022-02-15   
713       14667      Madrid                  Leganés  2022-04-24   

     gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
3            2.103                    NaN             NaN          2.223   
219          1.799                    NaN             NaN          1.929   
254          1.579                    NaN             NaN          1.721   
562          1.619                    NaN             NaN          1.739   
713          1.769                    NaN             NaN          1.909   

     gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
3               NaN     1.983       NaN           2.043        NaN        NaN   
219             NaN     1.849     1.469           1.919        NaN        NaN   
254             NaN     1.479       NaN           1.569        NaN        NaN   
562             NaN     1.497       NaN           1.557        NaN        NaN   
713             NaN     1.819       NaN           1.899        NaN        NaN   

     hydrogen  
3         NaN  
219       NaN  
254       NaN  
562       NaN  
713       NaN

## CSV 07

In [9]:
df_gasoline_07 = pd.read_csv('D:\IA\spanish_gas_prices_000000000007.csv')

In [10]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_07_v1 = df_gasoline_07[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5', 'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5', 
                                    'gasoline_98E10', 'diesel_A', 'diesel_B', 'diesel_premium', 
                                    'bioetanol', 'biodiesel', 'hydrogen']].copy()

In [11]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_07 = df_gasoline_07_v1[df_gasoline_07_v1["date"].str.contains("2022")]

In [12]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_07 = df_gasoline2022_07[(df_gasoline2022_07["region_name"]=="Madrid")]

In [13]:
df_madrid2022_07.head()

station_id region_name      municipality_name        date  \
3          10884      Madrid  Guadalix de la Sierra  2022-06-03   
658        13144      Madrid      Torrejón de Ardoz  2022-05-17   
868         4764      Madrid                 Madrid  2022-03-30   
1604        4622      Madrid                  Parla  2022-05-12   
1992        4647      Madrid              Alpedrete  2022-04-11   

      gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
3             2.159                    NaN             NaN          2.279   
658           1.939                    NaN             NaN            NaN   
868           1.899                    NaN             NaN          2.029   
1604          1.939                    NaN             NaN          2.089   
1992          1.809                  1.869             NaN            NaN   

      gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  \
3                NaN     2.009       NaN           2.069        NaN   
658              NaN     1.929       NaN           1.959        NaN   
868              NaN     1.929       NaN           1.939        NaN   
1604             NaN     1.939       NaN           2.029        NaN   
1992             NaN     1.849       NaN           1.939        NaN   

      biodiesel  hydrogen  
3           NaN       NaN  
658         NaN       NaN  
868         NaN       NaN  
1604        NaN       NaN  
1992        NaN       NaN

## CSV 08

In [14]:
df_gasoline_08 = pd.read_csv('D:\IA\spanish_gas_prices_000000000008.csv')

In [15]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_08_v1 = df_gasoline_08[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5', 'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5', 
                                    'gasoline_98E10', 'diesel_A', 'diesel_B', 
                                    'diesel_premium', 'bioetanol', 'biodiesel', 'hydrogen']].copy()

In [16]:
df_gasoline_08_v1.head()

station_id           region_name  municipality_name        date  \
0        6389       Castilla y León          Alcañices  2022-06-03   
1         842               Galicia         Pontevedra  2022-06-03   
2        3682  Comunidad Valenciana        Jávea/Xàbia  2022-05-31   
3        9445             Andalucia  Martín de la Jara  2022-05-31   
4        6731             Andalucia             Dúrcal  2022-05-31   

   gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0          2.119                    NaN             NaN          2.279   
1          2.129                    NaN             NaN          2.249   
2          2.069                    NaN             NaN          2.219   
3          1.959                    NaN             NaN          2.069   
4          2.049                    NaN             NaN          2.192   

   gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
0             NaN     1.999       NaN           2.099        NaN        NaN   
1             NaN     1.979       NaN           2.059        NaN        NaN   
2             NaN     1.909       NaN           1.999        NaN        NaN   
3             NaN     1.898       NaN           1.959        NaN        NaN   
4             NaN     1.899       NaN           1.989        NaN        NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [17]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_08 = df_gasoline_08_v1[df_gasoline_08_v1["date"].str.contains("2022")]

In [18]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_08 = df_gasoline2022_08[(df_gasoline2022_08["region_name"]=="Madrid")]

In [19]:
df_madrid2022_08.head()

station_id region_name municipality_name        date  gasoline_95E5  \
18        14808      Madrid            Madrid  2022-06-01          1.879   
22        10943      Madrid            Madrid  2022-06-03          2.184   
86         4461      Madrid       Majadahonda  2022-02-15          1.669   
195        4705      Madrid   Arganda del Rey  2022-02-23          1.669   
728        4545      Madrid      Nuevo Baztán  2022-02-15          1.672   

     gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  gasoline_98E10  \
18                     NaN             NaN          2.209             NaN   
22                     NaN             NaN            NaN             NaN   
86                     NaN             NaN          1.819             NaN   
195                    NaN             NaN            NaN             NaN   
728                    NaN             NaN            NaN             NaN   

     diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  hydrogen  
18      1.859       NaN           2.049        NaN        NaN       NaN  
22      2.029       NaN           2.095        NaN        NaN       NaN  
86      1.569       NaN           1.659        NaN        NaN       NaN  
195     1.559     1.019           1.569        NaN        NaN       NaN  
728     1.572       NaN           1.632        NaN        NaN       NaN

## CSV 09

In [20]:
df_gasoline_09 = pd.read_csv('D:\IA\spanish_gas_prices_000000000009.csv')

In [21]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_09_v1 = df_gasoline_09[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5', 
                                    'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5', 'gasoline_98E10', 
                                    'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol', 
                                    'biodiesel', 'hydrogen']].copy()

In [32]:
df_gasoline_09_v1.head()

station_id      region_name     municipality_name        date  \
0        6163  Castilla y León             Salamanca  2022-06-01   
1       11012  Castilla y León                 Boñar  2022-06-03   
2          63       País Vasco          Lasarte-Oria  2022-06-02   
3        2917           Madrid  Torres de la Alameda  2022-06-02   
4        3003         Cataluña             Barcelona  2022-06-02   

   gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0          2.104                    NaN             NaN          2.236   
1          2.129                  2.209             NaN          2.259   
2          2.139                    NaN             NaN          2.289   
3          2.089                    NaN             NaN          2.229   
4          2.109                    NaN             NaN          2.299   

   gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
0             NaN     1.919       NaN           2.019        NaN        NaN   
1             NaN     1.969     1.599           2.059        NaN        NaN   
2             NaN     1.979       NaN           2.049        NaN        NaN   
3             NaN     1.969       NaN           2.049        NaN        NaN   
4             NaN     1.959       NaN           2.029        NaN        NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [22]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_09 = df_gasoline_09_v1[df_gasoline_09_v1["date"].str.contains("2022")]

In [23]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_09 = df_gasoline2022_09[(df_gasoline2022_09["region_name"]=="Madrid")]

In [24]:
df_madrid2022_09.head()

station_id region_name           municipality_name        date  \
3          2917      Madrid        Torres de la Alameda  2022-06-02   
12         4576      Madrid                       Pinto  2022-06-02   
60         8458      Madrid           Torrejón de Ardoz  2022-04-04   
99         3077      Madrid  San Sebastián de los Reyes  2022-05-24   
270        4713      Madrid                    Alcorcón  2022-05-07   

     gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
3            2.089                    NaN             NaN          2.229   
12           2.129                  2.179             NaN            NaN   
60           1.854                    NaN             NaN          1.994   
99           1.999                    NaN             NaN          2.141   
270          1.829                    NaN             NaN          1.969   

     gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
3               NaN     1.969       NaN           2.049        NaN        NaN   
12              NaN     2.009     1.608           2.099        NaN        NaN   
60              NaN     1.925       NaN           2.005        NaN        NaN   
99              NaN     1.889       NaN           1.979        NaN        NaN   
270             NaN     1.859       NaN           1.939        NaN        NaN   

     hydrogen  
3         NaN  
12        NaN  
60        NaN  
99        NaN  
270       NaN

## CSV 10

In [26]:
df_gasoline_10 = pd.read_csv('D:\IA\spanish_gas_prices_000000000010.csv')

In [27]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_10_v1 = df_gasoline_10[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5', 'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5', 'gasoline_98E10', 
                                    'diesel_A', 'diesel_B', 'diesel_premium', 
                                    'bioetanol', 'biodiesel', 'hydrogen']].copy()

In [33]:
df_gasoline_10_v1.head()

station_id      region_name        municipality_name        date  \
0        2931         Cataluña                  Tordera  2022-05-31   
1        1865         Cataluña                 Terrassa  2022-05-31   
2       11008         Asturias                  Teverga  2022-06-02   
3       12572  Castilla y León                Salamanca  2022-06-01   
4       12968        Andalucia  Chiclana de la Frontera  2022-05-31   

   gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0          2.089                    NaN             NaN          2.221   
1          1.949                    NaN             NaN          2.129   
2          2.089                    NaN             NaN            NaN   
3          2.069                    NaN             NaN          2.199   
4          2.049                    NaN             NaN          2.179   

   gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
0             NaN     1.955     1.539           2.035        NaN        NaN   
1             NaN     1.834       NaN           1.904        NaN        NaN   
2             NaN     1.939     1.541           1.989        NaN        NaN   
3             NaN     1.919       NaN           1.999        NaN        NaN   
4             NaN     1.879       NaN           1.929        NaN        NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [28]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_10 = df_gasoline_10_v1[df_gasoline_10_v1["date"].str.contains("2022")]

In [29]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_10 = df_gasoline2022_10[(df_gasoline2022_10["region_name"]=="Madrid")]

In [30]:
df_madrid2022_10.head()

station_id region_name municipality_name        date  gasoline_95E5  \
13        13483      Madrid            Getafe  2022-05-31          2.023   
17        11472      Madrid            Madrid  2022-06-03          2.139   
19         3193      Madrid        Alcobendas  2022-06-02          2.089   
42         3110      Madrid             Parla  2022-02-01          1.629   
152        9167      Madrid   Arganda del Rey  2022-05-02          1.799   

     gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  gasoline_98E10  \
13                     NaN             NaN          2.149             NaN   
17                     NaN             NaN          2.299             NaN   
19                     NaN             NaN          2.239             NaN   
42                     NaN             NaN          1.769             NaN   
152                    NaN             NaN            NaN             NaN   

     diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  hydrogen  
13      1.925       NaN           1.995        NaN        NaN       NaN  
17      2.009       NaN           2.089        NaN        NaN       NaN  
19      1.939       NaN           2.029        NaN        NaN       NaN  
42      1.519       NaN           1.609        NaN        NaN       NaN  
152     1.849       NaN           1.859        NaN        NaN       NaN

## CSV 11

In [34]:
df_gasoline_11 = pd.read_csv('D:\IA\spanish_gas_prices_000000000011.csv')

In [35]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_11_v1 = df_gasoline_11[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5', 'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5', 'gasoline_98E10', 
                                    'diesel_A', 'diesel_B', 'diesel_premium', 
                                    'bioetanol', 'biodiesel', 'hydrogen']].copy()

In [36]:
df_gasoline_11_v1.head()

station_id region_name    municipality_name        date  gasoline_95E5  \
0          43  País Vasco   Arrazua-Ubarrundia  2022-06-03          2.029   
1        6781   Andalucia               Bailén  2022-06-02          2.069   
2        3168      Madrid  Pelayos de la Presa  2022-06-01          2.085   
3        2426    Cataluña   Sant Pere de Ribes  2022-06-01          2.159   
4         176  País Vasco    Amorebieta-Etxano  2022-06-01          2.099   

   gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  gasoline_98E10  \
0                    NaN             NaN          2.069             NaN   
1                    NaN             NaN          2.179             NaN   
2                    NaN             NaN          2.235             NaN   
3                    NaN             NaN          2.309             NaN   
4                    NaN             NaN            NaN             NaN   

   diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  hydrogen  
0     1.939       NaN           2.039        NaN        NaN       NaN  
1     1.969       NaN           2.039        NaN        NaN       NaN  
2     1.955       NaN           2.045        NaN        NaN       NaN  
3     1.979       NaN           2.069        NaN        NaN       NaN  
4     1.969       NaN           2.019        NaN        NaN       NaN

In [37]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_11 = df_gasoline_11_v1[df_gasoline_11_v1["date"].str.contains("2022")]

In [38]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_11 = df_gasoline2022_11[(df_gasoline2022_11["region_name"]=="Madrid")]

In [39]:
df_madrid2022_11.head()

station_id region_name    municipality_name        date  gasoline_95E5  \
2           3168      Madrid  Pelayos de la Presa  2022-06-01          2.085   
458        15009      Madrid    Alcalá de Henares  2022-03-16          1.889   
683         4328      Madrid                Pinto  2022-04-04          1.829   
886        10483      Madrid    Torrejón de Ardoz  2022-05-24          2.024   
1096        4713      Madrid             Alcorcón  2022-05-10          1.849   

      gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  gasoline_98E10  \
2                       NaN             NaN          2.235             NaN   
458                     NaN             NaN          2.039             NaN   
683                     NaN             NaN          1.959             NaN   
886                   2.089             NaN          2.156             NaN   
1096                    NaN             NaN          1.989             NaN   

      diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  hydrogen  
2        1.955       NaN           2.045        NaN        NaN       NaN  
458      1.879       NaN           1.969        NaN        NaN       NaN  
683      1.859       NaN           1.909        NaN        NaN       NaN  
886      1.909       NaN           2.019        NaN        NaN       NaN  
1096     1.869       NaN           1.949        NaN        NaN       NaN

## CSV 12

In [40]:
df_gasoline_12 = pd.read_csv('D:\IA\spanish_gas_prices_000000000012.csv')

In [41]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_12_v1 = df_gasoline_12[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5',
       'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5',
       'gasoline_98E10', 'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol',
       'biodiesel', 'hydrogen']].copy()

In [42]:
df_gasoline_12_v1.head()

station_id      region_name            municipality_name        date  \
0        6173  Castilla y León  Siete Iglesias de Trabancos  2022-05-31   
1        3092           Madrid                     Alcorcón  2022-06-03   
2        1854         Cataluña                      Manresa  2022-05-31   
3        7149        Andalucia                   Cantillana  2022-06-01   
4       11021  Castilla y León       Quintanilla de Onésimo  2022-05-31   

   gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0          2.034                  2.134             NaN            NaN   
1          2.119                    NaN             NaN          2.261   
2          2.089                    NaN             NaN          2.239   
3          2.029                  2.069             NaN            NaN   
4          2.019                    NaN             NaN          2.129   

   gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
0             NaN     1.899       NaN           1.979        NaN        NaN   
1             NaN     1.989       NaN           2.079        NaN        NaN   
2             NaN     1.939       NaN           2.029        NaN        NaN   
3             NaN     1.959       NaN           2.049        NaN        NaN   
4             NaN     1.869       NaN           1.919        NaN        NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [43]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_12 = df_gasoline_12_v1[df_gasoline_12_v1["date"].str.contains("2022")]

In [44]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_12 = df_gasoline2022_12[(df_gasoline2022_12["region_name"]=="Madrid")]

In [45]:
df_madrid2022_12.head()

station_id region_name municipality_name        date  gasoline_95E5  \
1          3092      Madrid          Alcorcón  2022-06-03          2.119   
19         3157      Madrid       Tres Cantos  2022-06-01          2.069   
104       14585      Madrid            Madrid  2022-01-20          1.437   
345        8445      Madrid          Alcorcón  2022-05-23          1.949   
432       14059      Madrid      Navalcarnero  2022-04-18          1.769   

     gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  gasoline_98E10  \
1                      NaN             NaN          2.261             NaN   
19                     NaN             NaN          2.211             NaN   
104                    NaN             NaN            NaN             NaN   
345                    NaN             NaN            NaN             NaN   
432                    NaN             NaN            NaN             NaN   

     diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  hydrogen  
1       1.989       NaN           2.079        NaN        NaN       NaN  
19      1.929       NaN           2.019        NaN        NaN       NaN  
104     1.307       NaN           1.327        NaN        NaN       NaN  
345     1.839       NaN           1.919        NaN        NaN       NaN  
432     1.789       NaN           1.829        NaN        NaN       NaN

## CSV 13

In [46]:
df_gasoline_13 = pd.read_csv('D:\IA\spanish_gas_prices_000000000013.csv')

In [47]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_13_v1 = df_gasoline_13[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5',
       'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5',
       'gasoline_98E10', 'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol',
       'biodiesel', 'hydrogen']].copy()

In [48]:
df_gasoline_13_v1.head()

station_id           region_name      municipality_name        date  \
0       11102           Extremadura           Almendralejo  2022-06-01   
1        3852  Comunidad Valenciana                Alfafar  2022-06-02   
2        7676              Canarias    San Miguel de Abona  2022-06-03   
3        5460    Castilla la Mancha             Dosbarrios  2022-06-03   
4        2230              Cataluña  Sant Cugat del Vallès  2022-06-01   

   gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0          2.049                    NaN             NaN            NaN   
1          2.109                    NaN             NaN          2.219   
2          1.495                    NaN             NaN          1.599   
3          2.139                    NaN             NaN          2.259   
4          2.139                    NaN             NaN          2.299   

   gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
0             NaN     1.949       NaN           2.019        NaN        NaN   
1             NaN     1.979       NaN           2.069        NaN        NaN   
2             NaN     1.499       NaN           1.609        NaN        NaN   
3             NaN     2.009       NaN           2.069        NaN        NaN   
4             NaN     1.989       NaN           2.069        NaN        NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [49]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_13 = df_gasoline_13_v1[df_gasoline_13_v1["date"].str.contains("2022")]

In [50]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_13 = df_gasoline2022_13[(df_gasoline2022_13["region_name"]=="Madrid")]

In [51]:
df_madrid2022_13.head()

station_id region_name        municipality_name        date  \
10        4349      Madrid                  Coslada  2022-05-31   
14        8458      Madrid        Torrejón de Ardoz  2022-06-01   
16        4536      Madrid                   Madrid  2022-06-02   
39        4580      Madrid  San Fernando de Henares  2022-05-31   
40       11167      Madrid                  Leganés  2022-06-02   

    gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
10          2.089                    NaN             NaN          2.219   
14          2.074                    NaN             NaN          2.214   
16          2.129                    NaN             NaN            NaN   
39          2.079                    NaN             NaN          2.219   
40          2.099                    NaN             NaN          2.241   

    gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
10             NaN     1.939       NaN           2.029        NaN        NaN   
14             NaN     1.949       NaN           2.029        NaN        NaN   
16             NaN     1.979       NaN           2.049        NaN        NaN   
39             NaN     1.919       NaN           2.009        NaN        NaN   
40             NaN     1.979       NaN           2.069        NaN        NaN   

    hydrogen  
10       NaN  
14       NaN  
16       NaN  
39       NaN  
40       NaN

## CSV 14

In [52]:
df_gasoline_14 = pd.read_csv('D:\IA\spanish_gas_prices_000000000014.csv')

In [53]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_14_v1 = df_gasoline_14[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5',
       'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5',
       'gasoline_98E10', 'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol',
       'biodiesel', 'hydrogen']].copy()

In [54]:
df_gasoline_14_v1.head()

station_id           region_name      municipality_name        date  \
0       13027       Castilla y León            Bañeza (La)  2022-06-02   
1        6810             Andalucia              Arquillos  2022-06-03   
2        8328                Madrid                Leganés  2022-06-03   
3        3505  Comunidad Valenciana                   Onda  2022-05-31   
4        5284    Castilla la Mancha  Tarazona de la Mancha  2022-06-01   

   gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0          2.109                    NaN             NaN            NaN   
1          2.069                    NaN             NaN            NaN   
2          2.099                    NaN             NaN          2.229   
3          2.004                  2.084             NaN          2.136   
4          2.069                  2.099             NaN            NaN   

   gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
0             NaN     1.979       NaN           2.049        NaN        NaN   
1             NaN     1.999     1.538           2.059        NaN        NaN   
2             NaN     1.979     1.629           2.079        NaN        NaN   
3             NaN     1.895       NaN           1.975        NaN        NaN   
4             NaN     1.939       NaN           1.989        NaN        NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [55]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_14 = df_gasoline_14_v1[df_gasoline_14_v1["date"].str.contains("2022")]

In [56]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_14 = df_gasoline2022_14[(df_gasoline2022_14["region_name"]=="Madrid")]

In [57]:
df_madrid2022_14.head()

station_id region_name   municipality_name        date  gasoline_95E5  \
2          8328      Madrid             Leganés  2022-06-03          2.099   
25         3200      Madrid         Torrelaguna  2022-05-31          2.089   
28         4535      Madrid              Madrid  2022-01-02          1.549   
162       15187      Madrid  Pozuelo de Alarcón  2022-03-25          1.819   
422        3129      Madrid        Ciempozuelos  2022-02-10          1.649   

     gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  gasoline_98E10  \
2                      NaN             NaN          2.229             NaN   
25                     NaN             NaN            NaN             NaN   
28                     NaN             NaN            NaN             NaN   
162                    NaN             NaN            NaN             NaN   
422                    NaN             NaN          1.759             NaN   

     diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  hydrogen  
2       1.979     1.629           2.079        NaN        NaN       NaN  
25      1.949       NaN           1.999        NaN        NaN       NaN  
28      1.419       NaN           1.485        NaN        NaN       NaN  
162     1.819       NaN           1.839        NaN        NaN       NaN  
422     1.539     1.150           1.589        NaN        NaN       NaN

## CSV 15

In [58]:
df_gasoline_15 = pd.read_csv('D:\IA\spanish_gas_prices_000000000015.csv')

In [59]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_15_v1 = df_gasoline_15[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5',
       'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5',
       'gasoline_98E10', 'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol',
       'biodiesel', 'hydrogen']].copy()

In [60]:
df_gasoline_15_v1.head()

station_id           region_name municipality_name        date  \
0        1823              Cataluña            Fortià  2022-06-03   
1       11598  Comunidad Valenciana           Torrent  2022-05-31   
2       11133              Cataluña              Rubí  2022-06-02   
3        8395              Cataluña     Molins de Rei  2022-06-02   
4        1259                Aragón          Zaragoza  2022-06-01   

   gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0          2.074                    NaN             NaN          2.214   
1          2.099                    NaN             NaN          2.239   
2          2.138                    NaN             NaN          2.198   
3          1.929                    NaN             NaN          2.069   
4          1.989                    NaN             NaN            NaN   

   gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
0             NaN     1.914       NaN           1.984        NaN        NaN   
1             NaN     1.939       NaN           2.009        NaN        NaN   
2             NaN     1.978       NaN           1.978        NaN        NaN   
3             NaN     1.899       NaN           1.979        NaN        NaN   
4             NaN     1.869     1.446           1.929        NaN        NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [61]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_15 = df_gasoline_15_v1[df_gasoline_15_v1["date"].str.contains("2022")]

In [62]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_15 = df_gasoline2022_15[(df_gasoline2022_15["region_name"]=="Madrid")]

In [63]:
df_madrid2022_15.head()

station_id region_name   municipality_name        date  gasoline_95E5  \
19          2912      Madrid              Madrid  2022-06-03          2.139   
529         4490      Madrid   Alcalá de Henares  2022-03-01          1.689   
1029        4714      Madrid      Colmenar Viejo  2022-04-08          1.779   
1312        3074      Madrid  Boadilla del Monte  2022-04-16          1.829   
1387        3136      Madrid           Valdemoro  2022-05-01          1.909   

      gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  gasoline_98E10  \
19                      NaN             NaN          2.299             NaN   
529                     NaN             NaN          1.839             NaN   
1029                    NaN             NaN          1.919             NaN   
1312                    NaN             NaN          1.969             NaN   
1387                    NaN             NaN          2.051             NaN   

      diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  hydrogen  
19       2.009       NaN           2.109        NaN        NaN       NaN  
529      1.599       NaN           1.689        NaN        NaN       NaN  
1029     1.819       NaN           1.899        NaN        NaN       NaN  
1312     1.879       NaN           1.949        NaN        NaN       NaN  
1387     1.989       NaN           2.079        NaN        NaN       NaN

## CSV 16

In [64]:
df_gasoline_16 = pd.read_csv('D:\IA\spanish_gas_prices_000000000016.csv')

In [65]:
# Se crea una tabla con solamente los datos que nos interesan
df_gasoline_16_v1 = df_gasoline_16[['station_id','region_name', 'municipality_name', 'date', 'gasoline_95E5',
       'gasoline_95E5_premium', 'gasoline_95E10', 'gasoline_98E5',
       'gasoline_98E10', 'diesel_A', 'diesel_B', 'diesel_premium', 'bioetanol',
       'biodiesel', 'hydrogen']].copy()

In [66]:
df_gasoline_16_v1.head()

station_id      region_name       municipality_name        date  \
0        6683        Andalucia                   Siles  2022-05-31   
1        6417  Castilla y León                   Soria  2022-06-03   
2        6185  Castilla y León         Sardón de Duero  2022-06-03   
3        6471        Andalucia            Dos Hermanas  2022-05-31   
4        7370        Andalucia  Roda de Andalucía (La)  2022-05-31   

   gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0          1.989                    NaN             NaN          2.079   
1          2.094                    NaN             NaN          2.236   
2          2.049                    NaN             NaN          2.159   
3          2.059                    NaN             NaN          2.179   
4          2.029                    NaN             NaN            NaN   

   gasoline_98E10  diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  \
0             NaN     1.899      1.49           1.909        NaN        NaN   
1             NaN     1.969       NaN           2.059        NaN        NaN   
2             NaN     1.909       NaN           1.929        NaN        NaN   
3             NaN     1.899       NaN           1.959        NaN        NaN   
4             NaN     1.859       NaN           1.919        NaN        NaN   

   hydrogen  
0       NaN  
1       NaN  
2       NaN  
3       NaN  
4       NaN

In [67]:
# Se filtra por la columna "date" para dar con las tomas hechas en 2022
df_gasoline2022_16 = df_gasoline_16_v1[df_gasoline_16_v1["date"].str.contains("2022")]

In [68]:
# Se filtra por la columna "region_name" para dar con aquellas estaciones registradas en Madrid
df_madrid2022_16 = df_gasoline2022_16[(df_gasoline2022_16["region_name"]=="Madrid")]

In [69]:
df_madrid2022_16.head()

station_id region_name  municipality_name        date  gasoline_95E5  \
26          3194      Madrid             Madrid  2022-06-02          2.164   
30          3204      Madrid   Orusco de Tajuña  2022-06-01          2.059   
158        12145      Madrid             Madrid  2022-03-07          1.667   
698         2906      Madrid        Moralzarzal  2022-05-05          1.919   
1001       10772      Madrid  Rivas-Vaciamadrid  2022-05-02          1.869   

      gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  gasoline_98E10  \
26                    2.244             NaN            NaN             NaN   
30                      NaN             NaN          2.169             NaN   
158                     NaN             NaN          1.779             NaN   
698                     NaN             NaN            NaN             NaN   
1001                    NaN             NaN          1.989             NaN   

      diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  hydrogen  
26       1.975       NaN           2.055        NaN        NaN       NaN  
30       1.939     1.533           1.989        NaN        NaN       NaN  
158      1.593       NaN           1.621        NaN        NaN       NaN  
698      1.979       NaN           2.025        NaN        NaN       NaN  
1001     1.929       NaN           1.959        NaN        NaN       NaN

Una vez se tienen todas la información de todos los archivos, las tablas resultantes se unirán en una misma, para comenzar a realizar el análisis.

In [170]:
df_gasoline2022 = pd.concat( [df_madrid2022_00, df_madrid2022_01, df_madrid2022_02, df_madrid2022_03, df_madrid2022_04, df_madrid2022_05, df_madrid2022_06, df_madrid2022_07, df_madrid2022_08, df_madrid2022_09, df_madrid2022_10, df_madrid2022_11, df_madrid2022_12, df_madrid2022_13, df_madrid2022_14, df_madrid2022_15, df_madrid2022_16], ignore_index=True )

In [171]:
df_gasoline2022.head()

station_id region_name       municipality_name        date  gasoline_95E5  \
0        4312      Madrid                  Madrid  2022-06-03          2.169   
1        3201      Madrid                  Madrid  2022-06-01          2.099   
2        3222      Madrid  Torrejón de la Calzada  2022-03-04          1.739   
3        4600      Madrid                  Madrid  2022-03-25          1.889   
4        4364      Madrid             Navacerrada  2022-02-25          1.664   

   gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  gasoline_98E10  \
0                  2.229             NaN            NaN             NaN   
1                    NaN             NaN          2.269             NaN   
2                  1.799             NaN            NaN             NaN   
3                    NaN             NaN          2.029             NaN   
4                  1.744             NaN          1.796             NaN   

   diesel_A  diesel_B  diesel_premium  bioetanol  biodiesel  hydrogen  
0     2.005       NaN           2.085        NaN        NaN       NaN  
1     1.959       NaN           2.049        NaN        NaN       NaN  
2     1.655       NaN           1.735        NaN        NaN       NaN  
3     1.919       NaN           1.999        NaN        NaN       NaN  
4     1.549       NaN           1.629        NaN        NaN       NaN

In [172]:
df_gasoline2022.shape

(95725, 15)

La tabla resultante muestra una cantidad de 95725 filas y 15 columnas con las que se trabajará a partir de ahora.

Se va a examinar los datos de cada columna, para poder ver si alguna de ellas contiene datos vacíos.

In [173]:
df_gasoline2022.isnull().sum()

station_id                   0
region_name                  0
municipality_name            0
date                         0
gasoline_95E5             2519
gasoline_95E5_premium    85415
gasoline_95E10           95598
gasoline_98E5            34976
gasoline_98E10           95725
diesel_A                  2304
diesel_B                 90584
diesel_premium           15464
bioetanol                95725
biodiesel                95456
hydrogen                 95725
dtype: int64

Columnas como "gasoline_98E10", "bioetanol" e "hydrogen" cuyos datos todos son null, por lo que se eliminarán.

In [174]:
df_gasoline2022 = df_gasoline2022.drop(["gasoline_98E10", "bioetanol", "hydrogen"], axis=1)

In [175]:
df_gasoline2022

station_id region_name       municipality_name        date  \
0            4312      Madrid                  Madrid  2022-06-03   
1            3201      Madrid                  Madrid  2022-06-01   
2            3222      Madrid  Torrejón de la Calzada  2022-03-04   
3            4600      Madrid                  Madrid  2022-03-25   
4            4364      Madrid             Navacerrada  2022-02-25   
...           ...         ...                     ...         ...   
95720       13893      Madrid                  Madrid  2022-01-02   
95721       14513      Madrid              Alcobendas  2022-01-24   
95722       11918      Madrid              Alcobendas  2022-01-07   
95723       14070      Madrid             Tres Cantos  2022-01-18   
95724       15161      Madrid       Torrejón de Ardoz  2022-01-08   

       gasoline_95E5  gasoline_95E5_premium  gasoline_95E10  gasoline_98E5  \
0              2.169                  2.229             NaN            NaN   
1              2.099                    NaN             NaN          2.269   
2              1.739                  1.799             NaN            NaN   
3              1.889                    NaN             NaN          2.029   
4              1.664                  1.744             NaN          1.796   
...              ...                    ...             ...            ...   
95720          1.459                    NaN             NaN            NaN   
95721          1.459                    NaN             NaN            NaN   
95722          1.459                    NaN             NaN          1.649   
95723          1.359                    NaN             NaN            NaN   
95724          1.379                    NaN             NaN            NaN   

       diesel_A  diesel_B  diesel_premium  biodiesel  
0         2.005       NaN           2.085        NaN  
1         1.959       NaN           2.049        NaN  
2         1.655       NaN           1.735        NaN  
3         1.919       NaN           1.999        NaN  
4         1.549       NaN           1.629        NaN  
...         ...       ...             ...        ...  
95720     1.299       NaN           1.339        NaN  
95721     1.339       NaN           1.359        NaN  
95722     1.329       NaN           1.359        NaN  
95723     1.229       NaN             NaN        NaN  
95724     1.237       NaN             NaN        NaN  

[95725 rows x 12 columns]

### Representación de gráficas

#### Media de cada carburante por mes

In [ ]:
df_gasoline2022_gasoline_95E5=df_gasoline2022[['date','gasoline_95E5']]
df_gasoline2022_gasoline_95E5_premium=df_gasoline2022[['date','gasoline_95E5_premium']]
df_gasoline2022_gasoline_95E10=df_gasoline2022[['date','gasoline_95E10']]
df_gasoline2022_gasoline_98E5=df_gasoline2022[['date','gasoline_98E5']]
df_gasoline2022_diesel_A=df_gasoline2022[['date','diesel_A']]
df_gasoline2022_diesel_B=df_gasoline2022[['date','diesel_B']]
df_gasoline2022_diesel_premium=df_gasoline2022[['date','diesel_premium']]
df_gasoline2022_biodiesel=df_gasoline2022[['date','biodiesel']]

In [ ]:
df_carburantes=[df_gasoline2022_gasoline_95E5,df_gasoline2022_gasoline_95E5_premium,df_gasoline2022_gasoline_95E10,df_gasoline2022_gasoline_98E5,df_gasoline2022_diesel_A,df_gasoline2022_diesel_B,df_gasoline2022_diesel_premium,df_gasoline2022_biodiesel]

#### Funcion que recibe como parametro un dataframe de un carburante y dibuja la grafica con las medias por mes (no es necesaria)

In [1]:
def grafica_media_mes(df_mes):
    
    df_mes['date']=pd.to_datetime(df_mes['date'])
    df_mes=df_mes.groupby(pd.Grouper(key="date", freq="1M")).mean()
    df_mes = df_mes.reset_index()
    df_mes=df_mes.replace({"2022-01-31":'Enero',"2022-02-28":'Febrero',"2022-03-31":'Marzo',"2022-04-30":'Abril',"2022-05-31":'Mayo',"2022-06-30":'Junio',"2022-07-31":'Julio',"2022-08-31":'Agosto', "2022-09-30":'Septiembre',"2022-10-31":'Octubre',"2022-11-30":'Noviembre',"2022-12-31":'Diciembre'})
    
    tipo_carburante=df_mes.columns[1]
    
    
    x = df_mes.date.values
    y = df_mes[tipo_carburante].values

    f, ax = plt.subplots(nrows=4, ncols=2,figsize=(8,8))
    plt.bar(x,y,color='0.9', edgecolor='blue', width=0.5)
    
    

    plt.ylim(0,2.5)
    plt.title("Media por mes Gasoline_95E5",fontsize=15)
    plt.xlabel("Carburante",fontsize=15)
    plt.ylabel("Euro/litro",fontsize=15)
    plt.yticks(fontsize=11)
    plt.xticks(fontsize=11)
    plt.show()


#### Funcion para filtrar los dataframe de los carburantes y filtrar por fechas

In [2]:
def filtrar_df(df_carburantes):
    nuevo_df_carburantes=[]
    for df_mes in df_carburantes:
        df_mes['date']=pd.to_datetime(df_mes['date'])
        df_mes=df_mes.groupby(pd.Grouper(key="date", freq="1M")).mean()
        df_mes = df_mes.reset_index()
        df_mes=df_mes.replace({"2022-01-31":'Enero',"2022-02-28":'Febrero',"2022-03-31":'Marzo',"2022-04-30":'Abril',"2022-05-31":'Mayo',"2022-06-30":'Junio',"2022-07-31":'Julio',"2022-08-31":'Agosto', "2022-09-30":'Septiembre',"2022-10-31":'Octubre',"2022-11-30":'Noviembre',"2022-12-31":'Diciembre'})
        nuevo_df_carburantes.append(df_mes)
        
    return nuevo_df_carburantes

In [ ]:
nuevo_df_carburantes=filtrar_df(df_carburantes)

##### Graficas de cada carburante con la media por mes

In [ ]:
tipo_carburante_1=nuevo_df_carburantes[0].columns[1] 
tipo_carburante_2=nuevo_df_carburantes[1].columns[1]    
tipo_carburante_3=nuevo_df_carburantes[2].columns[1]    
tipo_carburante_4=nuevo_df_carburantes[3].columns[1]    
tipo_carburante_5=nuevo_df_carburantes[4].columns[1]    
tipo_carburante_6=nuevo_df_carburantes[5].columns[1]    
tipo_carburante_7=nuevo_df_carburantes[6].columns[1]    
tipo_carburante_8=nuevo_df_carburantes[7].columns[1]    


x1 = nuevo_df_carburantes[0].date.values
y1= nuevo_df_carburantes[0][tipo_carburante_1].values

x2= nuevo_df_carburantes[1].date.values
y2= nuevo_df_carburantes[1][tipo_carburante_2].values

x3= nuevo_df_carburantes[2].date.values
y3= nuevo_df_carburantes[2][tipo_carburante_3].values

x4= nuevo_df_carburantes[3].date.values
y4= nuevo_df_carburantes[3][tipo_carburante_4].values

x5= nuevo_df_carburantes[4].date.values
y5= nuevo_df_carburantes[4][tipo_carburante_5].values

x6= nuevo_df_carburantes[5].date.values
y6= nuevo_df_carburantes[5][tipo_carburante_6].values

x7= nuevo_df_carburantes[6].date.values
y7= nuevo_df_carburantes[6][tipo_carburante_7].values

x8= nuevo_df_carburantes[7].date.values
y8= nuevo_df_carburantes[7][tipo_carburante_8].values


f, ax = plt.subplots(nrows=4, ncols=2,figsize=(15,15))

ax[0][0].bar(x1,y1,color='0.9', edgecolor='blue')
ax[0][0].set_title(tipo_carburante_1)


ax[1][0].bar(x2,y2, color='0.7', edgecolor='blue')
ax[1][0].set_title(tipo_carburante_2)

ax[2][0].bar(x3,y3, color='0.5', edgecolor='blue')
ax[2][0].set_title(tipo_carburante_3)

ax[3][0].bar(x4,y4 , color='0.3', edgecolor='blue')
ax[3][0].set_title(tipo_carburante_4)

ax[0][1].bar(x5,y5, color='0.1', edgecolor='blue')
ax[0][1].set_title(tipo_carburante_5)

ax[1][1].bar(x6,y6, color='1', edgecolor='blue')
ax[1][1].set_title(tipo_carburante_6)

ax[2][1].bar(x7,y7, color='0.2', edgecolor='blue')
ax[2][1].set_title(tipo_carburante_7)

ax[3][1].bar(x8,y8, color='0.8', edgecolor='blue')
ax[3][1].set_title(tipo_carburante_8)


plt.show()